bigtable-hbase-1.x-parent/bigtable-hbase-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java

```diff
@@ -348,23 +363,163 @@ public HBaseSchemaTranslator(SchemaTranslationOptions options) throws IOExceptio
+      this.mappingFilePath = mappingFilePath;
+      tableNameMappings = new HashMap<>();
+      tableScopedColumnFamilyMappings = new HashMap<>();
+    }
+
+    @VisibleForTesting
+    void parseMappingFile() throws FileNotFoundException {
+      Reader jsonReader = new FileReader(mappingFilePath);
+      Type mapType = new TypeToken<Map<String, String>>() {}.getType();
+      Map<String, String> schemaMapping = new Gson().fromJson(jsonReader, mapType);
+
+      if (schemaMapping == null) {
+        throw new IllegalStateException(
+            "SchemaMapping file does not contain valid schema mappings");
+      }
+
+      // Read the mapping file and organize the transformations in a structured way.
+      // Please note that we consciously chose a flat JSON file to make it easy for users to
+      // configure the schema translator.
+      for (Map.Entry<String, String> entry : schemaMapping.entrySet()) {
+        // Presence of a ":" indicates that the override is for a column family scoped to a table.
+        if (entry.getKey().contains(COLUMN_FAMILY_SEPARATOR)) {
+          // Split the key on separator. First part is the table and second part is column family.
+          String[] tableAndColumnFamily = entry.getKey().split(COLUMN_FAMILY_SEPARATOR, 2);
+          String table = tableAndColumnFamily[0];
+          String cf = tableAndColumnFamily[1];
+          tableScopedColumnFamilyMappings.putIfAbsent(table, new HashMap<>());
+          // entry<K,V>: key is the source table:cf and value is new cf.
+          tableScopedColumnFamilyMappings.get(tableAndColumnFamily[0]).put(cf, entry.getValue());
+        } else {
+          // Add the table mappings.
+          tableNameMappings.put(entry.getKey(), entry.getValue());
+        }
+      }
+    }
+
+    @Override
+    public ClusterSchemaDefinition transform(final ClusterSchemaDefinition originalSchema)
+        throws IOException, DeserializationException {
+      ClusterSchemaDefinition transformedSchema = new ClusterSchemaDefinition();
+
+      // Parse the mappings
+      parseMappingFile();
+
+      // Apply the transformations.
+      for (TableSchemaDefinition tableSchemaDefinition : originalSchema.tableSchemaDefinitions) {
+        String tableName = tableSchemaDefinition.name;
+        HTableDescriptor tableDescriptor = tableSchemaDefinition.getHbaseTableDescriptor();
+
+        // Override the table name if its present in the mapping file
+        if (tableNameMappings.containsKey(tableName)) {
+          tableName = tableNameMappings.get(tableName);
+        }
+        // Rename the table and copy all the other configs, including the column families.
+        HTableDescriptor newTableDescriptor =
+            new HTableDescriptor(TableName.valueOf(tableName), tableDescriptor);
+
+        // For all the column families that need renaming, replace them.
+        if (tableScopedColumnFamilyMappings.containsKey(tableSchemaDefinition.name)) {
```

Comment:
Please add a comment to make it explicit that you are using the original table name for renaming columns. Otherwise someone might optimize your code and replace `tableSchemaDefinition.name`  with `tableName`